lib/puppet_x/puppetlabs/cisco_ios/transport_shim.rb

```diff
@@ -301,18 +301,11 @@ class Puppet::ResourceApi::Transport::Wrapper
   attr_reader :transport, :schema
 
   def initialize(name, url_or_config_or_transport)
-    if url_or_config_or_transport.is_a? String
-      url = URI.parse(url_or_config_or_transport)
-      raise "Unexpected url '#{url_or_config_or_transport}' found. Only file:/// URLs for configuration supported at the moment." unless url.scheme == 'file'
-      raise "Trying to load config from '#{url.path}, but file does not exist." if url && !File.exist?(url.path)
-      config = self.class.deep_symbolize(Hocon.load(url.path, syntax: Hocon::ConfigSyntax::HOCON) || {})
-    elsif url_or_config_or_transport.is_a? Hash
-      config = url_or_config_or_transport
-    elsif transport_class?(name, url_or_config_or_transport)
+    if transport_class?(name, url_or_config_or_transport)
```

Comment:
Yea these changes will be breaking for Puppet 6.0.x which this file is meant to make compatible.

Reply:
I tested against Puppet 6 (6.0.3 and 6.4.2) and 5.5 locally and it functioned.

Tried to keep what I'm doing similar to what was there before.

Reply:
Yes but locally you probably had latest rsapi, on a PE with 6.0.x (2019.0) the install of rsapi is locked to 1.6.x of the rsapi and the required changes are in 1.8.x. so I wouldn't make the changes in this initialize

Reply:
The device.rb changes are acceptable

Reply:
Added check that the Puppet::Transport module exists before calling this code.

Previously it wasn't hit as the 'am I a hash' / 'am I a string' checks caught the input before it got to this 'if transport_class?' branch

Tested against RSAPI 1.6.4, Puppet 6.0.3
and latest RSAPI 1.8.1, Puppet 6.4.0